src/main/java/org/ossgang/commons/observable/Observables.java

```diff
@@ -31,10 +40,205 @@ private Observables() {
+     */
+    public static <V> ObservableValue<List<V>> zip(List<ObservableValue<V>> sources) {
+        return zip(sources, Function.identity());
+    }
+
+    /**
+     * Produces an {@link ObservableValue} that on each update of any source {@link ObservableValue} publishes the
+     * result of the combiner applied with the latest values of the other inputs.
+     * The order of the input values of the combiner is the same as the order of the provided source {@link ObservableValue}s
+     *
+     * @param sources  the input {@link ObservableValue}s
+     * @param combiner the combining function that will produce the result
+     * @param <V>      the input type
+     * @param <O>      the output type
+     * @return an {@link ObservableValue} that on each update of any source publishes the result of the
+     * combiner applied with the latest values of the other inputs.
+     */
+    public static <V, O> ObservableValue<O> combineLatest(List<ObservableValue<V>> sources,
+                                                          Function<List<V>, O> combiner) {
+        Property<O> mergedProperty = property();
+        Map<ObservableValue<V>, V> latestValues = new HashMap<>();
+        List<ObservableValue<V>> sourcesCopy = new ArrayList<>(sources);
+
+        for (ObservableValue<V> source : sourcesCopy) {
+            source.subscribe(sourceValue -> {
+                synchronized (latestValues) {
+                    latestValues.put(source, sourceValue);
+
+                    if (latestValues.keySet().containsAll(sourcesCopy)) {
+                        List<V> latestValueSnapshotInOrder = new ArrayList<>();
+                        for (ObservableValue<V> s : sourcesCopy) {
+                            latestValueSnapshotInOrder.add(latestValues.get(s));
+                        }
+
+                        mergedProperty.set(combiner.apply(latestValueSnapshotInOrder));
+                    }
+                }
+            }, FIRST_UPDATE);
+        }
+
+        return mergedProperty;
+    }
+
+    /**
+     * Produces an {@link ObservableValue} that on each update of any source {@link ObservableValue} publishes the
+     * result of applying the mapper {@link Function}. The {@link Map}s are used to avoid mathing values by index.
+     *
+     * @param sourcesMap the input {@link ObservableValue}s indexed
+     * @param mapper     the combining function that will produce the result
+     * @param <K>        the key for each input {@link ObservableValue} that is used for indexing
+     * @param <V>        the input type
+     * @param <O>        the output type
+     * @return an {@link ObservableValue} that on each update of any source {@link ObservableValue} publishes the
+     * result of applying the mapper {@link Function}. The {@link Map}s are used to avoid mathing values by index.
+     */
+    public static <K, V, O> ObservableValue<O> combineLatest(Map<K, ObservableValue<V>> sourcesMap, Function<Map<K, V>, O> mapper) {
```

Comment:
Shall we use the same concept as for zip() here, i.e. delegate from the List to the Map case?